From 2095d5e610e665dcb9d166a0531c63625db53026 Mon Sep 17 00:00:00 2001
From: Tam Nguyen <tam.nguyen.xa@renesas.com>
Date: Thu, 14 Nov 2024 18:26:36 +0700
Subject: [PATCH] arm64: dts: renesas: r8a779g0-native: Add Native device
 support

In the default environment, all UIO devices were enabled, Native
devices (such as DU devices, GPU, ...) that was conflicted with
UIO devices.

The patch will create a new Device Tree to support enabling Native
devices that support for display output, GPU, and basic peripherals.

For convenience to use:
- "r8a779g{0, 3, 4, 5,}-whitehawk.dtb": All UIO devices enable, Native devices
  disabled.
- "r8a779g{0, 3, 4, 5}-whitehawk-native.dtb": All UIO devices enable except for
  display
output, GPU and basic peripherals.

Signed-off-by: Tam Nguyen <tam.nguyen.xa@renesas.com>
---
 arch/arm64/boot/dts/renesas/Makefile          |  5 ++
 .../dts/renesas/r8a779g0-whitehawk-native.dts | 19 +++++
 .../dts/renesas/r8a779g3-whitehawk-native.dts | 20 +++++
 .../dts/renesas/r8a779g4-whitehawk-native.dts | 20 +++++
 .../dts/renesas/r8a779g5-whitehawk-native.dts | 20 +++++
 .../dts/renesas/white-hawk-native-common.dtsi | 81 +++++++++++++++++++
 .../boot/dts/renesas/white-hawk-native.dtsi   | 39 +++++++++
 7 files changed, 204 insertions(+)
 create mode 100644 arch/arm64/boot/dts/renesas/r8a779g0-whitehawk-native.dts
 create mode 100644 arch/arm64/boot/dts/renesas/r8a779g3-whitehawk-native.dts
 create mode 100644 arch/arm64/boot/dts/renesas/r8a779g4-whitehawk-native.dts
 create mode 100644 arch/arm64/boot/dts/renesas/r8a779g5-whitehawk-native.dts
 create mode 100644 arch/arm64/boot/dts/renesas/white-hawk-native-common.dtsi
 create mode 100644 arch/arm64/boot/dts/renesas/white-hawk-native.dtsi

diff --git a/arch/arm64/boot/dts/renesas/Makefile b/arch/arm64/boot/dts/renesas/Makefile
index 40bff86de3e7..447b7127eed2 100644
--- a/arch/arm64/boot/dts/renesas/Makefile
+++ b/arch/arm64/boot/dts/renesas/Makefile
@@ -74,6 +74,11 @@ dtb-$(CONFIG_ARCH_R8A779A0) += r8a779a0-falcon.dtb
 dtb-$(CONFIG_ARCH_R8A779G0) += r8a779g0-whitehawk.dtb
 dtb-$(CONFIG_ARCH_R8A779G0) += r8a779g0-whitehawk-sound.dtb
 
+dtb-$(CONFIG_ARCH_R8A779G0) += r8a779g0-whitehawk-native.dtb
+dtb-$(CONFIG_ARCH_R8A779G0) += r8a779g3-whitehawk-native.dtb
+dtb-$(CONFIG_ARCH_R8A779G0) += r8a779g4-whitehawk-native.dtb
+dtb-$(CONFIG_ARCH_R8A779G0) += r8a779g5-whitehawk-native.dtb
+
 dtb-$(CONFIG_ARCH_R8A779G0) += r8a779g3-white-hawk.dtb
 dtb-$(CONFIG_ARCH_R8A779G0) += r8a779g3-whitehawk-sound.dtb
 
diff --git a/arch/arm64/boot/dts/renesas/r8a779g0-whitehawk-native.dts b/arch/arm64/boot/dts/renesas/r8a779g0-whitehawk-native.dts
new file mode 100644
index 000000000000..8e5565e8bae1
--- /dev/null
+++ b/arch/arm64/boot/dts/renesas/r8a779g0-whitehawk-native.dts
@@ -0,0 +1,19 @@
+// SPDX-License-Identifier: (GPL-2.0-only OR BSD-2-Clause)
+/*
+ * Device Tree Source for the R-Car V4H White Hawk board
+ * Support Native driver.
+ *
+ * Copyright (C) 2024 Renesas Electronics Corp.
+ */
+
+/dts-v1/;
+#include "white-hawk-native.dtsi"
+#include "r8a779g0.dtsi"
+#include "white-hawk-cpu-common.dtsi"
+#include "white-hawk-common.dtsi"
+#include "white-hawk-native-common.dtsi"
+
+/ {
+	model = "Renesas White Hawk board based on r8a779g0";
+	compatible = "renesas,whitehawk", "renesas,r8a779g0";
+};
diff --git a/arch/arm64/boot/dts/renesas/r8a779g3-whitehawk-native.dts b/arch/arm64/boot/dts/renesas/r8a779g3-whitehawk-native.dts
new file mode 100644
index 000000000000..031922443c5c
--- /dev/null
+++ b/arch/arm64/boot/dts/renesas/r8a779g3-whitehawk-native.dts
@@ -0,0 +1,20 @@
+// SPDX-License-Identifier: (GPL-2.0-only OR BSD-2-Clause)
+/*
+ * Device Tree Source for the R-Car V4H-7 White Hawk board
+ * Support Native driver.
+ *
+ * Copyright (C) 2024 Renesas Electronics Corp.
+ */
+
+/dts-v1/;
+#include "white-hawk-native.dtsi"
+#include "r8a779g3.dtsi"
+#include "white-hawk-cpu-common.dtsi"
+#include "white-hawk-common.dtsi"
+#include "white-hawk-native-common.dtsi"
+
+/ {
+	model = "Renesas White Hawk board based on r8a779g0";
+	compatible = "renesas,whitehawk", "renesas,r8a779g0",
+				"renesas,r8a779g3";
+};
diff --git a/arch/arm64/boot/dts/renesas/r8a779g4-whitehawk-native.dts b/arch/arm64/boot/dts/renesas/r8a779g4-whitehawk-native.dts
new file mode 100644
index 000000000000..52e037f63136
--- /dev/null
+++ b/arch/arm64/boot/dts/renesas/r8a779g4-whitehawk-native.dts
@@ -0,0 +1,20 @@
+// SPDX-License-Identifier: (GPL-2.0-only OR BSD-2-Clause)
+/*
+ * Device Tree Source for the R-Car V4H-5 White Hawk board
+ * Support Native driver.
+ *
+ * Copyright (C) 2024 Renesas Electronics Corp.
+ */
+
+/dts-v1/;
+#include "white-hawk-native.dtsi"
+#include "r8a779g4.dtsi"
+#include "white-hawk-cpu-common.dtsi"
+#include "white-hawk-common.dtsi"
+#include "white-hawk-native-common.dtsi"
+
+/ {
+	model = "Renesas White Hawk board based on r8a779g0";
+	compatible = "renesas,whitehawk", "renesas,r8a779g0",
+				"renesas,r8a779g4";
+};
diff --git a/arch/arm64/boot/dts/renesas/r8a779g5-whitehawk-native.dts b/arch/arm64/boot/dts/renesas/r8a779g5-whitehawk-native.dts
new file mode 100644
index 000000000000..227eba0b8e6f
--- /dev/null
+++ b/arch/arm64/boot/dts/renesas/r8a779g5-whitehawk-native.dts
@@ -0,0 +1,20 @@
+// SPDX-License-Identifier: (GPL-2.0-only OR BSD-2-Clause)
+/*
+ * Device Tree Source for the R-Car V4H-3 White Hawk board
+ * Support Native driver.
+ *
+ * Copyright (C) 2024 Renesas Electronics Corp.
+ */
+
+/dts-v1/;
+#include "white-hawk-native.dtsi"
+#include "r8a779g5.dtsi"
+#include "white-hawk-cpu-common.dtsi"
+#include "white-hawk-common.dtsi"
+#include "white-hawk-native-common.dtsi"
+
+/ {
+	model = "Renesas White Hawk board based on r8a779g0";
+	compatible = "renesas,whitehawk", "renesas,r8a779g0",
+				"renesas,r8a779g5";
+};
diff --git a/arch/arm64/boot/dts/renesas/white-hawk-native-common.dtsi b/arch/arm64/boot/dts/renesas/white-hawk-native-common.dtsi
new file mode 100644
index 000000000000..43670d563ea9
--- /dev/null
+++ b/arch/arm64/boot/dts/renesas/white-hawk-native-common.dtsi
@@ -0,0 +1,81 @@
+// SPDX-License-Identifier: (GPL-2.0-only OR BSD-2-Clause)
+/*
+ * Device Tree Source support Native device
+ * for the White Hawk CPU and Breakout boards
+ *
+ * Copyright (C) 2024 Renesas Electronics Corp.
+ */
+
+&i2c1 {
+	sn65dsi86_0@2c {
+		status = "okay";
+	};
+
+	max96789@60 {
+		status = "okay";
+	};
+
+	gmsl0: gmsl-deserializer@49 {
+		status = "okay";
+		i2c-mux {
+			i2c@0 {
+				status = "okay";
+			};
+
+			i2c@1 {
+				status = "okay";
+			};
+
+			i2c@2 {
+				status = "okay";
+			};
+
+			i2c@3 {
+				status = "okay";
+			};
+		};
+	};
+
+	gmsl1: gmsl-deserializer@4b {
+		status = "okay";
+		i2c-mux {
+			i2c@0 {
+				status = "okay";
+			};
+
+			i2c@1 {
+				status = "okay";
+			};
+
+			i2c@2 {
+				status = "okay";
+			};
+
+			i2c@3 {
+				status = "okay";
+			};
+		};
+	};
+};
+
+&i2c4 {
+	sn65dsi86_1@2c {
+		status = "okay";
+	};
+};
+
+&du {
+	status = "okay";
+};
+
+&dsi0 {
+	status = "okay";
+};
+
+&dsi1 {
+	status = "okay";
+};
+
+&dsc {
+	status = "okay";
+};
\ No newline at end of file
diff --git a/arch/arm64/boot/dts/renesas/white-hawk-native.dtsi b/arch/arm64/boot/dts/renesas/white-hawk-native.dtsi
new file mode 100644
index 000000000000..84ade98e4b34
--- /dev/null
+++ b/arch/arm64/boot/dts/renesas/white-hawk-native.dtsi
@@ -0,0 +1,39 @@
+// SPDX-License-Identifier: (GPL-2.0-only OR BSD-2-Clause)
+/*
+ * Device Tree Source support Native for the White Hawk CPU
+ * boards with R-Car V4H
+ *
+ * Copyright (C) 2024 Renesas Electronics Corp.
+ */
+
+#include "r8a779g0.dtsi"
+
+/ {
+	compatible = "renesas,r8a779g0-native", "renesas,r8a779g0";
+
+	soc: soc {
+		/delete-node/du0;
+		/delete-node/du_01;
+		/delete-node/dsi0;
+		/delete-node/dsi0_01;
+		/delete-node/dsi0_02;
+		/delete-node/dsi0_03;
+		/delete-node/dsi0_04;
+		/delete-node/dsi0_05;
+		/delete-node/dsi0_06;
+		/delete-node/dsi0_07;
+		/delete-node/dsi0_08;
+		/delete-node/dsi1;
+		/delete-node/dsi1_01;
+		/delete-node/dsi1_02;
+		/delete-node/dsi1_03;
+		/delete-node/dsi1_04;
+		/delete-node/dsi1_05;
+		/delete-node/dsi1_06;
+		/delete-node/dsi1_07;
+		/delete-node/dsi1_08;
+		/delete-node/doc@feba0000;
+		/delete-node/doc0_01;
+		/delete-node/dsc@feb8d000;
+	};
+};
\ No newline at end of file
-- 
2.34.1

